Adyen/Core/Core Protocols/Component.swift

```diff
@@ -14,10 +14,15 @@ extension Component {
 
     /// Finalizes the payment if there is any, after being processed by payment provider.
     /// - Parameter success: The status of the payment.
-    /// :nodoc:
-    public func finalizeIfNeeded(with success: Bool) {
-        (self as? FinalizableComponent)?.didFinalize(with: success)
+    /// - Parameter completion: The block to execute after the component finalize it's activity.
+    /// Use of this block is recommended for ApplePayComponent. You may specify nil for this parameter.
+    public func finalizeIfNeeded(with success: Bool, completion: (() -> Void)?) {
         stopLoadingIfNeeded()
```

Comment:
Small difference but would changing the order of `stopLoadingIfNeeded` and `finalize(...)` cause any issue